include/spotify/json/encode.hpp

```diff
@@ -38,16 +40,18 @@ json_never_inline std::string encode(const value_type &value) {
   return encode(default_codec<value_type>(), value);
 }
 
-template <typename storage_type = std::string, typename codec_type>
-json_never_inline encoded_value<storage_type> encode_value(const codec_type &codec, const typename codec_type::object_type &object) {
+template <typename codec_type>
+json_never_inline encoded_value encode_value(
+    const codec_type &codec,
+    const typename codec_type::object_type &object) {
   encode_context context;
   codec.encode(context, object);
-  return encoded_value<storage_type>(context.data(), context.size(), typename encoded_value<storage_type>::unsafe_unchecked());
+  return encoded_value(std::move(context), encoded_value::unsafe_unchecked());
```

Comment:
This is a big win with the new code, that we can move the `encode_context` into the `encoded_value` instead of making a copy of the JSON data.